app/workers/scheduled_publishing_worker.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class ScheduledPublishingWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 10
+  sidekiq_retry_in { 30 }
+
+  def perform(id)
+    Edition.transaction do
+      edition = Edition.lock.find(id)
```

Comment:
Not sure if this is a great suggestion of mine or not - as it may well conflict with part of a suggestion here: https://trello.com/c/VrE9kdK4/744-we-make-unnecessary-references-to-document-in-our-controllers

But to avoid having to worry about the RecordNotFound exception we could do `find_by(id: id)` which would return nil instead of raising.

Reply:
I've changed this to `find_by(id: id)`.  I've kept the lock, is that correct?